### PR TITLE
[Evoker] Implement Source of Magic & Potent Mana modules

### PR DESCRIPTION
**Pull request by Krealle**
### Description
Implemented statistics modules for the shared Evoker talents `Source of Magic` and `Potent Mana`.

Percentage for `Source of Magic` mana gain is just the amount of a mana bar it has provided.

Enabled for all specs, tagging @trevorm4 since it includes Preservation.

- Test report URL: `/report/ZdMWQn9DvKr8HVJB/9-Mythic+Fyrakk+the+Blazing+-+Kill+(9:38)/Vollmer/standard/statistics`
- Screenshot(s):
image
image

### CODE REVIEW

**Comment by trevorm4** [comment]
non wcl-query code looks good, I'll let topple or emallson approve that part. I know that we use LazyLoadStatistics for ones that rely on WCL queries in general since they're relatively expensive and not everybody cares about that stat all the time.

File: src/analysis/retail/evoker/shared/modules/talents/PotentMana.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        const filter = `
+        ((source.name='${window.player}' OR source.owner.name='${window.player}') AND
+        timestamp>=${window.start - this.owner.fight.start_time} AND 
+        timestamp<=${window.end - this.owner.fight.start_time})
+        `;
+        return filter;
+      })
+      .join(' OR ');
+
+    filter = String(filter) + 'AND effectiveHealing > 0';
```
Comment (ToppleTheNun) [inline src/analysis/retail/evoker/shared/modules/talents/PotentMana.tsx:49]: I think you may want an extra space before the `AND` here?

**Review by ToppleTheNun — APPROVED** [review]
Looks good as a whole, one very small note on the query itself.

File: src/analysis/retail/evoker/shared/modules/talents/PotentMana.tsx (hunk 1 above)
Comment (Krealle) [reply src/analysis/retail/evoker/shared/modules/talents/PotentMana.tsx:49]: filter output looks like this
```
((source.name='Vìdel' OR source.owner.name='Vìdel') AND
timestamp>=0 AND 
timestamp<=578671)
AND effectiveHealing > 0
```
So shouldn't need an extra space.

**Review by emallson — APPROVED** [review]
queries look sane to me

**Comment by emallson** [comment]
we do sometimes fire off extra queries in normal operation for a spec. the brew code does that for Purifying Brew. as long as it is relatively cheap (like constant number of queries done) and used sparingly it is fine.

**Comment by trevorm4** [comment]
fixed in #6648

**Pull request merged**